typescript-local-storage/data.ts

```diff
@@ -0,0 +1,18 @@
+/* exported todos */
+interface Todo {
+  todoId: string;
+  task: string;
+  isCompleted: boolean;
+}
+
+let todos: Todo[] = [];
+
+window.addEventListener('beforeunload', function () {
+  const JSONtodos = JSON.stringify(todos);
+  localStorage.setItem('javascript-local-storage', JSONtodos);
+
+  const previousJSONTodos = localStorage.getItem('javascript-local-storage');
+  if (previousJSONTodos !== null) {
+    todos = JSON.parse(previousJSONTodos);
+  }
+});
```

Comment:
You are setting and getting your `todos` in your `beforeunload`. The issue with this is that you will only ever get the new data when you about about to unload the page. You want this to happen when the page loads.